src/open_inwoner/openklant/models.py

```
@@ -213,13 +212,11 @@ def api_url(self):
         return urlunparse((scheme, netloc, path, params, query, fragment))
 
     @classmethod
-    def from_django_settings(cls) -> Self:
+    def from_django_settings(cls) -> Self | None:
         from django.conf import settings
 
         if not (config := getattr(settings, "OPENKLANT2_CONFIG", None)):
-            raise ImproperlyConfigured(
-                "Please set OPENKLANT2_CONFIG in your settings to configure OpenKlant2"
-            )
+            return None
```

I think it's still clearer to keep this: it's the core purpose of the method, and if the settings are wrong, we should raise here and the caller should handle it (in this case, `OpenKlant2Service` could just let it bubble up I think without catching it explicitly).

---

I was thinking about bringing the `OpenKlant2Service` in line with the `eSuiteService` (the config for the latter does not raise an exception, it's the job of the service). However, it's probably better to leave as is and rethink this when we swap out the config-via-settings with a proper config model

---

Sounds good.